lib/iex/lib/iex/dochelp.ex

```diff
@@ -0,0 +1,49 @@
+defmodule IEx.DocHelp do
+  @moduledoc """
+  Documents the callbacks required for a DocHelp module.
+
+  This interface consists of the functions
+
+  * `documentation(module)`
+
+  * `documentation(module, function)`
+
+  * `documentation(module, function, arity)`
+
+  All of these functions return a tuple of status and doc_list.
+
+  * `{:found, doc_list}`     - Documentation was found
+
+  * `{:not_found, doc_list}` - Documentation was not found and
+    this helper is sure it can't be found
+
+  * `{:unknown, doc_list}`   - Helper does not know how to find docs for
+    arguments given.
+
+  doc_list consists of a list of `{header, markdown}` tuples for each
+  item found.
+
+  The module also includes some helper functions for writing
+  DocHelp implementations.
+  """
+
+  @type header :: String.t
+  @type body :: String.t
+  @type doc_tuple :: {header, body}
+  @type doc_list :: [doc_tuple]
+  @type doc_return :: {:found, doc_list} | {:not_found, doc_list} | {:unknown, doc_list}
+
+  @callback documentation(module) :: doc_return
+  @callback documentation(module, function) :: doc_return
+  @callback documentation(module, function, arity) :: doc_return
+
+  @doc """
+  Checks whether `Code.get_docs` should have documentation for the module.
+  """
+  def has_get_docs?(module) do
```

Comment:
This function name is strange since it contains two verbs (has and get)... Maybe consider something like `can_get_docs?`